tests/test_mpi.py

```diff
@@ -820,26 +820,46 @@ def test_poke_progress(self):
 
         trees = retrieve_iteration_tree(op._func_table['compute0'].root)
         assert len(trees) == 2
-        tree = trees[1]
-        # Make sure `pokempi0` is within the outer Iteration
+        tree = trees[0]
+        # Make sure `pokempi0` is the last node within the outer Iteration
         assert len(tree) == 2
         assert len(tree.root.nodes) == 2
-        call = tree.root.nodes[0]
+        call = tree.root.nodes[1]
         assert call.name == 'pokempi0'
         assert call.arguments[0].name == 'msg0'
+        try:
+            # W/ OpenMP, we prod until all comms have completed
+            assert call.then_body[0].body[0].is_While
+            # W/ OpenMP, we expect dynamic thread scheduling
+            assert 'dynamic,1' in tree.root.pragmas[0].value
+            assert configuration['openmp']
+        except AttributeError:
+            # W/o OpenMP, it's a different story
+            assert call._single_thread
+            assert not configuration['openmp']
 
         # Now we do as before, but enforcing loop blocking (by default off,
         # as heuristically it is not enabled when the Iteration nest has depth < 3)
         op = Operator(eqn, dle=('advanced', {'blockinner': True}))
         trees = retrieve_iteration_tree(op._func_table['bf0'].root)
         assert len(trees) == 2
         tree = trees[1]
-        # Make sure `pokempi0` is within the inner Iteration over blocks
-        assert len(tree) == 4
+        # Make sure `pokempi0` is the last node within the inner Iteration over blocks
+        assert len(tree) == 2
         assert len(tree.root.nodes[0].nodes) == 2
-        call = tree.root.nodes[0].nodes[0]
+        call = tree.root.nodes[0].nodes[1]
         assert call.name == 'pokempi0'
         assert call.arguments[0].name == 'msg0'
+        try:
```

Comment:
I don't really like try except in tests

Reply:
mm agreed, but do you have alternatives?

Reply:
```
if configuration['openmp']
   ...
else:
  ...
```

or make `isopenmp` a fixture maybe.

Reply:
given the assert within try/except (see `assert configuration ...`),  it's practically the same....
... BUT you're right, your way is definitely clearer. I'll change it